src/core/public/application/application_service.tsx

```diff
@@ -364,3 +369,10 @@ const updateStatus = <T extends AppBase>(app: T, statusUpdaters: AppUpdaterWrapp
     ...changes,
   };
 };
+
+function relativeToAbsolute(url: string) {
+  // convert all link urls to absolute urls
+  const a = document.createElement('a');
+  a.setAttribute('href', url);
+  return a.href;
+}
```

Comment:
So, this implementation can seems strange, as it relies on actual DOM `a` node behavior, however
- This is the same approach that is currently used in `NavLinks` to creates the absolute url
- The behavior is specified
- This just works (tm), when other snippets to properly parse and convert a relative url to an absolute one are 'way' more complex, and requires to parse `window.location` to construct the url.

Still, tell me if we want to change the implementation